server/auth/gatekeeper.go

```diff
@@ -370,3 +371,10 @@ func DefaultClientForAuthorization(authorization string) (*rest.Config, *servert
 		Kubernetes:  kubeClient,
 	}, nil
 }
+
+func mergeServerRestConfig(argoServerConfig *rest.Config, newConfig *rest.Config) *rest.Config {
+	newConfig.Burst = argoServerConfig.Burst
+	newConfig.QPS = argoServerConfig.QPS
+	// TO DO: Merge other common configurations，such as RateLimiter.
```

Comment:
the user agent was the other one I [mentioned](https://github.com/argoproj/argo-workflows/pull/12575#discussion_r1473515842) from the Server. that's the only other one currently I think, no?